doc/user_guide.md

```diff
@@ -0,0 +1,341 @@
+
+```SQL
+TRAIN model_identifier
+WITH
+  model_attr_expr [, model_attr_expr ...]
+  [, train_attr_expr ...]
+```
+
+- *model_identifier* indicates the model type. e.g. `DNNClassifier`. Please refer to [Models](#models) for a complete list of supported models.
+- *model_attr_expr* indicates the model attribute. e.g. `n_classes = 3`. Please refer to [Models](#models) for details.
+- *train_attr_expr* indicates the training attribute. e.g. `EPOCHS = 10`. Please refer to [Hyperparameters](#hyperparameters) for details.
+
+For example, if you wanna train a DNNClassifier, which has 2 hidden layers and each layer has 10 hidden units, with 10 epochs, you can write
+
+```SQL
+SELECT ...
+TRAIN DNNClassifer
+WITH
+  hidden_units = [10, 10],
+  n_classes = 3,
+  EPOCHS = 10
+...
+```
+
+### Column clause
+
+The *column clause* indicates the field name to be used as training features, along with their preprocessing if needed, e.g. `COLUMN sepal_length, sepal_width, petal_length, petal_width`.
+
+```SQL
+COLUMN column_expr [, column_expr ...]
+  | COLUMN column_expr [, column_expr ...] FOR column_name
+    [COLUMN column_expr [, column_expr ...] FOR column_name ...]
+```
+
+- *column_expr* indicates the field name and the preprocessing method on the field content. e.g. `sepal_length`, `NUMERIC(dense, 3)`. Please refer to [Feature columns](#feature-columns) for preprocessing details.
+- *column_name* indicates the feature column names for the model inputs. Some models such as [DNNLinearCombinedClassifier](https://www.tensorflow.org/api_docs/python/tf/estimator/DNNLinearCombinedClassifier) have`linear_feature_columns` and `dnn_feature_columns` as feature column input.
+
+For example, if you wanna use field `sepal_length`, `sepal_width`, `petal_length`, and `petal_width` as the features, without any preprocessing on the field content, you can write
+
+```SQL
+SELECT ...
+TRAIN ...
+COLUMN sepal_length, sepal_width, petal_length, petal_width
+...
+```
+
+### Label clause
+
+The *label clause* indicates the field name to be used as the training label, along with their preprocessing if needed, e.g. `LABEL class`.
+
+```SQL
+LABEL label_expr
+```
+
+- *label_expr* indicates the field name and the preprocessing method on the field content. e.g. `class`.
+
+Note: some field name may look like SQLFlow keywords. For example, the table may contain a field named label. You can use double quotes around the name `LABEL "label"` to work around the parsing error.
+
+### Save clause
+
+The *save clause* indicates the table name to save the trained model
+
+```SQL
+INTO table_references
+```
+
+- *table_references* indicates the table to save the trained model. e.g. `sqlflow_model.my_dnn_model`.
+
+Note: SQLFlow team is actively working on supporting saving model to third-party storage services such as AWS S3, Google Storage and Alibaba OSS.
+
+## Feature columns
```

Comment:
Actually, feature columns descriptions also include descriptions of feeding and processing input data (delimiter, shape).